# SPDX-License-Identifier: GPL-3.0-or-later
# Copyright (C) 2026 moneroism
# Alpine's musl-macros.patch hunk for src/elfcompress.c, refreshed for elfutils 0.196
# (0.196 added the dirname declaration where the hunk used to apply).

--- a/src/elfcompress.c
+++ b/src/elfcompress.c
@@ -41,6 +41,14 @@
 /* Really should come from libgen.h, but we poisoned basename in system.h.  */
 extern char *dirname(char *path);
 
+#if !defined(FNM_EXTMATCH)
+# define FNM_EXTMATCH 0
+#endif
+
+#if !defined(ALLPERMS)
+# define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO) /* 07777 */
+#endif
+
 /* Name and version of program.  */
 ARGP_PROGRAM_VERSION_HOOK_DEF = print_version;
 
